pandas/core/exchange/buffer.py

```diff
@@ -0,0 +1,75 @@
+from typing import (
+    Optional,
+    Tuple,
+)
+
+import numpy as np
+
+from pandas.core.exchange.dataframe_protocol import (
+    Buffer,
+    DlpackDeviceType,
+)
+
+
+class PandasBuffer(Buffer):
+    """
+    Data in the buffer is guaranteed to be contiguous in memory.
+    """
+
+    def __init__(self, x: np.ndarray, allow_copy: bool = True) -> None:
+        """
+        Handle only regular columns (= numpy arrays) for now.
+        """
+        if not x.strides == (x.dtype.itemsize,):
+            # The protocol does not support strided buffers, so a copy is
+            # necessary. If that's not allowed, we need to raise an exception.
+            if allow_copy:
+                x = x.copy()
+            else:
+                raise RuntimeError(
+                    "Exports cannot be zero-copy in the case "
+                    "of a non-contiguous buffer"
+                )
+
+        # Store the numpy array in which the data resides as a private
+        # attribute, so we can use it to retrieve the public attributes
+        self._x = x
+
+    @property
+    def bufsize(self) -> int:
+        """
+        Buffer size in bytes.
+        """
+        return self._x.size * self._x.dtype.itemsize
+
+    @property
+    def ptr(self) -> int:
+        """
+        Pointer to start of the buffer as an integer.
+        """
+        return self._x.__array_interface__["data"][0]
+
+    def __dlpack__(self):
+        """
+        DLPack not implemented in NumPy yet, so leave it out here.
+        """
+        raise NotImplementedError("__dlpack__")
```

Comment:
Should this be implemented since NumPy added support for dlpack in https://github.com/numpy/numpy/pull/19083 ?

Reply:
It should, so I've stubbed a simple implementation.